Modules.md

```diff
@@ -170,30 +217,45 @@ reserve up to the *maximum* is not. When a maximum memory size is *not* declared
 on architectures with limited virtual address space, engines should allocate
 only the initial size and reallocate on demand.
 
-The initial contents of linear memory are zero by default. However, the memory
-section contains a possibly-empty array of *segments* (analogous to `.data`)
-which can specify the initial contents of fixed `(offset, length)` ranges of
-memory.
-
-The linear memory section may optionally declare that the instance's
-linear memory is *externally aliasable*. How linear memory is aliased is up
-to the host environment (as with all module exports). The
-[Web](Web.md#aliasing-linear-memory-from-JS) would reflect exported linear
-memory to JS as an `ArrayBuffer`. The MVP does not currently provide for
-*importing* linear memory though this may be added
-[in the future](FutureFeatures.md#importing-linear-memory).
+## Data section
+
+The initial contents of linear memory are zero. The *data section* contains a
+possibly-empty array of *data segments* which specify the initial contents
+of fixed `(offset, length)` ranges of a given memory, specified by its [linear
+memory index](#index-space). This section is analogous to the `.data` section
+of native executables.
+
+## Table section
+
+A *table section* may contain zero or more definitions of distinct 
+[tables](AstSemantics.md#tables) which are added to the 
+[table index space](#index-spaces). Each table section declares an *element
+type*, *initial length*, and optional *maximum length*.
+
+In the MVP, tables can only be resized via host-defined APIs (such as
+the JavaScript [`WebAssembly.Table.prototype.grow`](JS.md#webassemblytableprototypegrow)).
+A `grow_table` may be added in the [future](FutureFeatures.md#more-table-operators-and-types).
+In either case, table growth is guaranteed to fail if attempting to grow past
+the declared maximum. As with linear memory, when a maximum is declared,
+implementations *should* (non-normative) attempt to reserve virtual memory up to
+the maximum size. While failure to allocate the *initial* memory size is a
+runtime error, failure to reserve up to the *maximum* is not. When a maximum
+memory size is *not* declared, on architectures with limited virtual address
+space, engines should allocate only the initial size and reallocate on demand.
+
+## Elements section
+
+For function tables, the intial contents of the tables' elements are sentinel
+values that throw if called. The *elements section* contains a possibly-empty
+array of *element segments* which define the initial contents of fixed
+`(offset, length)` ranges of a given table, specified by its 
+[table index](#index-spaces). The elements are specified by their 
+[function index](#index-spaces). This section is the table analogue of the
+linear memory [data section](#data-section).
 
```

Comment:
Sorry, I don't understand what the "elements section" contains? What are the "element segments"? Are they sub-ranges of table index spaces? Or is this a section with all the elements from all tables with the elements grouped into ranges as in the tables? Why is this needed and how is it to be used?